### PR TITLE
Add user controller tests

### PR DESCRIPTION
**Pull request by siemensikkema**
- changed UserAuthenticating static functions to non static ones for
  better testability
- renamed "login" and "logout" to "logIn" and "logOut"

### CODE REVIEW

File: Sources/JWTKeychain/Controllers/API/UserController.swift [hunk 1]
```diff
@@ -32,46 +35,48 @@ open class UserController<U: UserAuthenticating>: UserControllerType {
...
+        let user = try userAuthenticator.logIn(request: request)
         let token = try user.createToken(using: signer)
         return try makeResponse(token: token, user: user)
     }
 
-    open func logout(request: Request) throws -> ResponseRepresentable {
-        _ = try U.logOut(request: request)
+    open func logOut(request: Request) throws -> ResponseRepresentable {
+        _ = try userAuthenticator.logOut(request: request)
+        // TODO: is it really necessary to return this response? A status code of 200 should suffice.
```
Comment (steffendsommer) [inline Sources/JWTKeychain/Controllers/API/UserController.swift]: This is probably inspired by how we do APIs at Nodes (https://github.com/nodes-vapor/readme/blob/master/Documentation/how-to-write-apis.md) as we promise our frontends always to return a response.

**Review by steffendsommer — APPROVED** [review]

File: Sources/JWTKeychain/Controllers/API/UserController.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: ah but in that case, shouldn't it be a 'status' key?

```
Endpoints should always return a JSON payload.
When an endpoint doesn't have meaningful data to return (e.g. when deleting something), use a status key to communicate the status of the endpoint.
```

Comment (steffendsommer) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: Yes, it should

Comment (siemensikkema) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: So `["status": "success"]`? And what about the one for reset password email? `["status": "Instructions were sent to the provided email"]`? Or the same as logOut? Different key?

Comment (steffendsommer) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: Let's always use the `status` key - I'm ok with the `resetPasswordEmail` describing that the instructions has been sent.

Comment (siemensikkema) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: I've pushed a new commit to implement this

Comment (steffendsommer) [reply Sources/JWTKeychain/Controllers/API/UserController.swift]: LGTM 👍

**Pull request merged**